File: VMS_Backend/VMS_Backend/Data/DatabaseModels/Geofence.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VMS_Backend.Data.DatabaseModels
{
    [Table("geofence")]
    public class Geofence
    {
        [Key]
        [Column("id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        
        [Column("name")]
        [MaxLength(50)]
        public string Name { get; set; }
        
        [Column("company_id")] 
        public int CompanyId { get; set; }
        [ForeignKey("CompanyId")]
        public Company Company { get; set; } 
        
        [Column("coords")]
        public string Coords { get; set; }
        
        [Column("is_leave_restricted")]
        public bool IsLeaveRestricted { get; set; }
        
        [Column("is_enter_restricted")]
        public bool IsEnterRestricted { get; set; }
    }
}

File: VehicleMonitoringSystemWeb/vms_frontend/src/models/geofence.ts
import GeoCoordinate from "./geoCoordinate";

export default class Geofence {
  public id: number|undefined;
  public companyId: number;
  public name: string;
  public coords: GeoCoordinate[];
  public isLeaveRestricted: boolean;
  public isEnterRestricted: boolean;
  public color: string;

  constructor(id: number|undefined, companyId: number, name: string, coords: GeoCoordinate[],
              isLeaveRestricted: boolean, isEnterRestricted: boolean, color: string) {
    this.id = id;
    this.companyId = companyId;
    this.name = name;
    this.coords = coords;
    this.isLeaveRestricted = isLeaveRestricted;
    this.isEnterRestricted = isEnterRestricted;
    this.color = color;
  }
}
